test/functional/runner/index.js

```diff
@@ -425,52 +469,57 @@ function testOperation(operation, obj, context, options) {
   const operationName = translateOperationName(operation.name);
 
   if (operation.arguments) {
-    Object.keys(operation.arguments).forEach(key => {
-      if (key === 'callback') {
-        args.push(() =>
-          testOperations(operation.arguments.callback, context, { swallowOperationErrors: false })
-        );
-        return;
-      }
-
-      if (['filter', 'fieldName', 'document', 'documents', 'pipeline'].indexOf(key) !== -1) {
-        return args.unshift(operation.arguments[key]);
-      }
+    args = resolveOperationArgs(operationName, operation.arguments, context);
+
+    if (args == null) {
+      args = [];
+      Object.keys(operation.arguments).forEach(key => {
```

Comment:
Can we get rid of this now that we have `resolveOperationArgs`?

Reply:
Unfortunately not yet, since `resolveOperationArgs` currently only supports the `distinct` operation. I was trying to reduce the amount of change in this PR.